src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -43,5 +64,24 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 throw new ArgumentException(ex.Message, symbol.ToDisplayString());
             }
         }
+
+        private static ConstructorInitializerSyntax GenerateBaseConstructorInitializer(ImmutableArray<IMethodSymbol> baseTypeConstructors)
+        {
+            ConstructorInitializerSyntax constructorInitializer = SyntaxFactory.ConstructorInitializer(SyntaxKind.BaseConstructorInitializer);
+
+            IMethodSymbol baseConstructor = baseTypeConstructors.OrderBy(c => c.Parameters.Count()).First();
+            foreach (IParameterSymbol parameter in baseConstructor.Parameters)
+            {
+                IdentifierNameSyntax identifier;
+                if (parameter.Type.IsValueType)
+                    identifier = SyntaxFactory.IdentifierName("default");
+                else
+                    identifier = SyntaxFactory.IdentifierName("default!");
```

Comment:
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/7412651/216306938-fcfb99fd-7b86-40cf-bffa-2ee62365655d.png">

This might be a the right algorithm to identify when to use `default` vs `default!`.